main.py

```diff
@@ -260,49 +360,9 @@ def main(cl_arguments):
                           to_train, opt_params, schd_params,
                           args.shared_optimizer, args.load_model, phase="main")
 
-    # Select model checkpoint from main training run to load
-    if not args.do_target_task_training:
-        log.info("In strict mode because do_target_task_training is off. "
-                 "Will crash if any tasks are missing from the checkpoint.")
-        strict = True
-    else:
-        strict = False
-
-    if args.do_target_task_training and not args.allow_reuse_of_pretraining_parameters:
-        # If we're training models for evaluation, which is always done from scratch with a fresh
-        # optimizer, we shouldn't load parameters for those models.
-        # Usually, there won't be trained parameters to skip, but this can happen if a run is killed
-        # during the do_target_task_training phase.
-        task_names_to_avoid_loading = [task.name for task in target_tasks]
-    else:
-        task_names_to_avoid_loading = []
-
-    if not args.load_eval_checkpoint == "none":
-        # This is to load a particular eval checkpoint.
-        log.info("Loading existing model from %s...", args.load_eval_checkpoint)
-        load_model_state(model, args.load_eval_checkpoint,
-                         args.cuda, task_names_to_avoid_loading, strict=strict)
-    else:
-        # Look for eval checkpoints (available only if we're restoring from a run that already
-        # finished), then look for training checkpoints.
-
-        if args.transfer_paradigm == "finetune":
```

Comment:
We've moved this into the loop over all the target tasks, so we're now saving a new model for each target task when we should not be (this model has been fine-tuned for a particular target task and shouldn't be used for another task).

Reply:
Oh sorry! I misread the code down below I think.